config/recipes/elastic-agent/fleet-kubernetes-integration-noroot.yaml

```
@@ -0,0 +1,299 @@
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: manage-agent-hostpath-permissions
+spec:
+  selector:
+    matchLabels:
+      name: manage-agent-hostpath-permissions
+  template:
+    metadata:
+      labels:
+        name: manage-agent-hostpath-permissions
+    spec:
+      # This is only required when running in an selinux-enabled/Openshift environment.
+      # Ensure this user has been added to the privileged scc in the correct namespace.
+      # oc adm policy add-scc-to-user privileged -z elastic-agent -n elastic-apps
+      # serviceAccountName: elastic-agent
+      volumes:
+        - hostPath:
+            path: /var/lib/elastic-agent
+            type: DirectoryOrCreate
+          name: "agent-data"
+      initContainers:
+        - name: manage-agent-hostpath-permissions
+          # UBI is only required when needing the `chcon` binary when running
+          # in an senlinux-enabled/Openshift environment. If that
+          # is not required then the following smaller image can be used instead:
+          # image: registry.access.redhat.com/ubi8/ubi-minimal:latest
+          image: docker.io/bash:5.2.15
+          resources:
+            limits:
+              cpu: 100m
+              memory: 32Mi
+          securityContext:
+            # privileged is only required when running in an selinux-enabled/Openshift environment.
+            # privileged: true
+            runAsUser: 0
+          volumeMounts:
+            - mountPath: /var/lib/elastic-agent
+              name: agent-data
+          command:
+          - 'bash'
+          - '-e'
+          - '-c'
+          - |-
+            # Adjust this be /var/lib/elastic-agent/YOUR-NAMESPACE/YOUR-AGENT-NAME/state
+            # Multiple directories are supported for the fleet-server + agent use case.
+            dirs=(
+              "/var/lib/elastic-agent/default/elastic-agent/state"
+              "/var/lib/elastic-agent/default/fleet-server/state"
+              )
+            for dir in ${dirs[@]}; do
+              if [[ ! -d "${dir}" ]]
+              then
+                echo "creating ${dir} directory"
+                mkdir -p "${dir}"
+              fi
```

With `-p`, no error will be reported if a directory given as an operand already exists. 
```suggestion
              mkdir -p "${dir}"
```